### ISSUE TITLE
Would it be possible to adjust this software to work with other fantech mouses?

### ISSUE THREAD

**Issue by brunobmello25** [GuessWhatBBQ/FantechX9ThorDriver#10]
I have just bought a fantech aria xd7 and would love to be able to use it on linux as well. I would guess the only variables that would need tweaking would be the self.x9_vendorid and the self.x9_productid (which are currently hardcoded in the backend python file), right?

I am willing to help doing some testing and maybe I'll open a PR if that's okay

**Comment by GuessWhatBBQ** [comment]
Yes of course. PRs are welcome. Unfortunately I didn't design this to be configurable with other mouses. Ideally the user should be able to configure the vendor and product ids from the GUI. Implementing that wasn't on my mind initially as I only later realized the same code could be used on other mouses after user reports.

I dont use this mouse anymore and am thus incapable of testing any changes, so you'd have to ensure any changes don't affect the X9 Thor users

**Comment by brunobmello25** [comment]
Yeah I'd have some trouble guaranteeing that the changes I make don't break the X9 Thor users experience :/ maybe if someone that uses this mouse shows up here I can get their help to test it? 

For now I'll see if I can focus on getting it working with my Aria XD7. I'll update this thread as I go

**Comment by muhhae** [comment]
Just wondering is there any  progress on this? And also where you get that bytecode and if possible can i have reference that you use? thanks

**Comment by brunobmello25** [comment]
@muhhae not really on my part, no =/

I got busy with some other projects (both work and personal) and ended up not working on this. I would really like to get it working on linux though, so if you need any help let me know.

What do you mean by bytecode?

**Comment by GuessWhatBBQ** [comment]
I reverse engineered the bytecodes using Wireshark and a lot of patience so I think this information isn't readily available anywhere

**Comment by muhhae** [comment]
> I reverse engineered the bytecodes using Wireshark and a lot of patience so I think this information isn't readily available anywhere

Alright Ok then, because on me, the bytecode, I think is slightly different, thanks.

**Comment by 0xwagmi** [comment]
Easiest way to find  the vendor_id and the  pid is , type `lsusb` command it will display your mouse 

eg: 

`Bus 001 Device 002: ID 18f8:0fc0 [Maxxter] USB GAMING MOUSE 
`    
 18f8 -vendor_id 
 0fc0 -pid

**Comment by TheMrYah** [comment]
@Oxwagmi No, that will not work. Those are not in hex format. You can use this code: 
### list_usb.py

`import usb.core, usb.util
devs = usb.core.find(find_all=True)
for d in devs:
    try:
        man = usb.util.get_string(d, d.iManufacturer) or ""
        prod = usb.util.get_string(d, d.iProduct) or ""
    except Exception:
        man = prod = "<strings-unavailable>"
    print(f"{hex(d.idVendor)}:{hex(d.idProduct)}  vendor_dec={d.idVendor} prod_dec={d.idProduct}  -> {man} / {prod}")
`

You can make a function to auto-detect it when the program is executed. 
And to test it, use this code, just adjust the value:

### test_find.py

`import usb.core
dev = usb.core.find(idVendor=0x30fa, idProduct=0x1701)
print("found:", bool(dev))
print(dev)`

**I would really like it if someone made a new fork from this with all these shits, and dpi + other functionality.**

**Comment by sabahmax-dev** [comment]
Thanks for offering to help! I've opened PR #15 to add support for other Fantech mice, including the Aria XD7, by making the vendor and product IDs configurable. This should fix the issue of the driver only working with the hardcoded Fantech X9 Thor IDs.
